timemachine/fe/bar.py

```diff
@@ -196,24 +202,24 @@ def bootstrap_bar(
         )
         bootstrap_samples.append(bar_result)
 
-    return full_bar_result, np.array(bootstrap_samples)
+    return full_bar_result, full_bar_err, np.array(bootstrap_samples)
 
 
 def bar_with_bootstrapped_uncertainty(
     u_kln: NDArray, n_bootstrap=100, maximum_iterations: int = DEFAULT_MAXIMUM_ITERATIONS
 ) -> Tuple[float, float]:
     """Given 2-state u_kln, returns free energy difference and uncertainty computed by bootstrapping."""
 
-    df, bootstrap_dfs = bootstrap_bar(u_kln, n_bootstrap=n_bootstrap, maximum_iterations=maximum_iterations)
+    df, ddf, bootstrap_dfs = bootstrap_bar(u_kln, n_bootstrap=n_bootstrap, maximum_iterations=maximum_iterations)
 
     # warn if bootstrap distribution deviates significantly from normality
     normaltest_result = normaltest(bootstrap_dfs)
     pvalue_threshold = 1e-3  # arbitrary, small
     if normaltest_result.pvalue < pvalue_threshold:
         logger.warning(f"bootstrapped errors non-normal: {normaltest_result}")
 
-    # regardless, summarize as if normal
-    ddf = np.std(bootstrap_dfs)
+    # Take the max of the full error estimate and the bootstrapped error. Summarize as if normal regardless
+    ddf = max(ddf, np.std(bootstrap_dfs))
```

Comment:
@maxentile suggested taking the max of the full bar error vs the bootstrapped error

Reply:
Changed in https://github.com/proteneer/timemachine/pull/1324/commits/5fecf5756257fbf59b4f55294c011dd6eb549030 to do `np.maximum`. Also per offline changed to call `np.nan_to_num(ddf)` so that values are set to zero https://github.com/proteneer/timemachine/pull/1324/commits/beecb3914eec19822486ed0e1ff8792f568cc4eb.